app/src/main/java/nu/yona/app/ui/dashboard/DayActivityDetailFragment.java

```diff
@@ -98,148 +99,131 @@ public void onClick(View v)
-							&& (visibleItemCount + firstVisibleItemPosition) >= totalItemCount)
-					{
-						loadMoreItems();
-					}
-				}
-			}
+			loadMoreItems(nestedScrollView);
 		}
 	};
 
-	private void loadMoreItems()
+	private void loadMoreItems(NestedScrollView nestedScrollView)
 	{
-		fetchComments(viewPager.getCurrentItem());
+		View view = nestedScrollView.getChildAt(nestedScrollView.getChildCount() - 1);
+		int diff = (view.getBottom() - (nestedScrollView.getHeight() + nestedScrollView.getScrollY()));
+		if (diff == 0)
+		{
+			int visibleItemCount = mLayoutManager.getChildCount();
+			int totalItemCount = mLayoutManager.getItemCount();
+			int firstVisibleItemPosition = mLayoutManager.findFirstVisibleItemPosition();
+			EmbeddedYonaActivity embeddedYonaActivity = YonaApplication.getEventChangeManager().getDataState().getEmbeddedDayActivity();
+			if (isViewScrollable(embeddedYonaActivity, visibleItemCount, firstVisibleItemPosition, totalItemCount))
+			{
+				fetchComments(viewPager.getCurrentItem());
+			}
+		}
 	}
 
+	private boolean isViewScrollable(EmbeddedYonaActivity embeddedYonaActivity, int visibleItemCount, int firstVisibleItemPosition, int totalItemCount)
+	{
+		return (embeddedYonaActivity != null && embeddedYonaActivity.getPage() != null
+				&& embeddedYonaActivity.getPage().getNumber() < embeddedYonaActivity.getPage().getTotalPages()
+				&& (visibleItemCount + firstVisibleItemPosition) >= totalItemCount);
+	}
 
 	@Override
 	public void onCreate(@Nullable Bundle savedInstanceState)
 	{
 		super.onCreate(savedInstanceState);
-		if (getArguments() != null)
+		if (getArguments() == null)
 		{
-			if (getArguments().get(AppConstant.YONA_BUDDY_OBJ) != null)
-			{
-				if (getArguments().get(AppConstant.YONA_BUDDY_OBJ) instanceof YonaBuddy)
-				{
-					yonaBuddy = (YonaBuddy) getArguments().get(AppConstant.YONA_BUDDY_OBJ);
-				}
-			}
-			if (getArguments().getSerializable(AppConstant.YONA_THEME_OBJ) != null)
-			{
-				mYonaHeaderTheme = (YonaHeaderTheme) getArguments().getSerializable(AppConstant.YONA_THEME_OBJ);
-			}
+			return;
+		}
+		if (getArguments().get(AppConstant.YONA_BUDDY_OBJ) instanceof YonaBuddy)
+		{
+			yonaBuddy = (YonaBuddy) getArguments().get(AppConstant.YONA_BUDDY_OBJ);
+		}
+		if (getArguments().getSerializable(AppConstant.YONA_THEME_OBJ) != null)
+		{
+			mYonaHeaderTheme = (YonaHeaderTheme) getArguments().getSerializable(AppConstant.YONA_THEME_OBJ);
 		}
 	}
 
 
 	@Nullable
 	@Override
-	public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState)
+	public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState)
+	{
+		initializeViewAttributes(inflater);
+		initializeCommentControl(view);
+		if (getArguments() != null)
+		{
+			if (getArguments().get(AppConstant.OBJECT) instanceof DayActivity)
+			{
+				activity = (DayActivity) getArguments().get(AppConstant.OBJECT);
```

Comment:
This argument lookup construct pretty common. I suggest adding the below two methods to ``BaseFragment``:
```
	protected <T> T getArgument(String key, Class<T> type)
	{
		return getArgument(key, type, null);
	}

	protected <T> T getArgument(String key, Class<T> type, T defaultValue)
	{
		Bundle arguments = getArguments();
		if (arguments == null)
		{
			return defaultValue;
		}
		Object argument = arguments.get(key);
		if (type.isInstance(argument))
		{
			return type.cast(argument);
		}
		return defaultValue;
	}
```
With that, the entire if-statement to assign the activity if given as argument can be replace with this:
```
activity = getArgument(AppConstant.OBJECT, DayActivity.class, activity);
```

In other places, you just want to get the argument value and null if it's not there. That's what the first method does:
```
activity = getArgument(AppConstant.OBJECT, DayActivity.class);
```

Note that the above construct can be used to replace calls to ``Object get(String key)``, but also to ``Serializable getSerializable(String key)`` and ``String getString(String key)``.